apps/backend/src/modules/tasks/task.service.ts

```diff
@@ -104,6 +121,23 @@ class TaskService implements Service {
 		return tasks.map((task) => task.toObject());
 	}
 
+	public async getNotesByTaskId(taskId: number): Promise<TaskNoteDto[]> {
+		const task = await this.taskRepository.find(taskId);
+
+		if (!task) {
+			throw new TaskError({
+				message: ErrorMessage.TASK_NOT_FOUND,
+				status: HTTPCode.NOT_FOUND,
+			});
+		}
```

Comment:
I think this check is redundant, if there's no task, then the notes array will be empty